verified_boot_app_spec.md

```diff
@@ -0,0 +1,634 @@
+the user decides to trust for given bootloader.
+
+If the bootloader is signed only by Microsoft keys/certificates the trust
+prompt is deferred for later and the application attemps to process next
+bootloader if available. Trust prompt for Microsoft keys/certificates has the
+lowest priority (ignoring `BootOrder`) in such case. The application will scan
+the current partition for files that may be alternative usable bootloaders. It
+searches recursively for files using simple heuristics, such as file name:
+`bootx64.efi`, `elilo.efi`, `grubx64.efi`, `shimx64.efi` and `bootmgfw.efi` or
+`.efi` file extension matching. If any of the files contains a signature not
+made by Microsoft keys, they are prioritized for trust prompts. Once all other
+options are exhausted, the application will prompt to trust the Microsoft
+keys/certificates if any bootloaders signed by them are found.
+
+The application may continue booting the currently processed bootloader or
+proceed to check next bootloader and continue with trust prompts until all
+options are exhausted or user decides to continue booting.
+
+Only the `db` (permitted image signers) may be updated during this phase. No
+changes to `PK` are performed until the end of the provisioning workflow. This
+restriction ensures that the platform's trust anchors and ownership state are
+not altered prematurely.
+
+At the end of the provisioning workflow (once trusted key database is
+configured), the application creates ephemeral Platform Key `PK`, discards its
+private part, and enrolls the public part into `PK` variable. Enrolling the
+`PK` while UEFI Secure Boot is in Setup Mode will cause the firmware to
+transition the UEFI Secure Boot to User Mode and activate enforced image
+verification (enable UEFI Secure Boot).
+
+### 3.5 Boot Option Modifications
+
+When trusted key database is configured. `PK` is enrolled and UEFI Secure Boot
+is enabled, the application will proceed with booting the selected option.
+However, if user chose to trust a key/certificate of an option that was not the
+top priority in the `BootOrder` variable, the application will update the
+`BootOrder` variable to ensure the trusted bootloader is prioritized
+appropriately.
+
+When the application is not launched for the first time, it is in the
+interactive menu mode. The menu may offer a possibility to add new options
+from discovered bootloaders and modify bootorder if the user decides to extend
+the trusted key database by new keys/certificate or bootloaders.
+
+## 4. Security Model
+
+The application adheres to the security model defined in the UEFI Secure Boot
+architecture. The application is designed with adhering to the following
+rules:
+
+* No key is trusted without explicit user action. The rule applies only to the
+  bootloader executed directly by the application. If the bootloader
+  introduces another key into the trust chain, it is out of scope of the
+  application.
```

Comment:
Some of this is redundant with the third bullet point.

```suggestion
* No key is trusted without explicit user action. The rule applies only to the
  bootloader executed directly by the application.
```

Reply:
```suggestion
* No key is trusted without explicit user action. The rule applies only to the
  bootloader executed directly by the application or firmware.
```

Once we are provisioned, we can leave the booting process to the firmware, so the firmware will be booting the entry next time.